rskj-core/src/main/java/co/rsk/core/RskFactory.java

```diff
@@ -210,15 +210,26 @@ public JsonRpcWeb3ServerHandler getJsonRpcWeb3ServerHandler(Web3 web3Service, Rs
     }
 
     @Bean
-    public Web3WebSocketServer getWeb3WebSocketServer(RskSystemProperties rskSystemProperties,
-                                                      JsonRpcWeb3ServerHandler serverHandler) {
+    public Web3WebSocketServer getWeb3WebSocketServer(
+            RskSystemProperties rskSystemProperties,
+            Ethereum ethereum,
+            JsonRpcWeb3ServerHandler serverHandler,
+            JsonRpcSerializer serializer) {
+        EthSubscriptionNotificationEmitter emitter = new EthSubscriptionNotificationEmitter(ethereum, serializer);
+        RskJsonRpcHandler jsonRpcHandler = new RskJsonRpcHandler(emitter, serializer);
         return new Web3WebSocketServer(
-            rskSystemProperties.rpcWebSocketBindAddress(),
-            rskSystemProperties.rpcWebSocketPort(),
-            serverHandler
+                rskSystemProperties.rpcWebSocketBindAddress(),
+                rskSystemProperties.rpcWebSocketPort(),
+                jsonRpcHandler,
+                serverHandler
         );
     }
 
+    @Bean
+    public JsonRpcSerializer getJsonRpcSerializer() {
+        return new JsonRpcSerializer();
```

Comment:
I would let `JsonRpcSerializer` to be an interface and the actual implementation to be `JacksonBasedRpcSerializer`. I'm not against injecting the `ObjectMapper` directly in the `EthSubscriptionNotificationEmitter` though